ios/brave-ios/Sources/Growth/LanguageMetrics.swift

```diff
@@ -0,0 +1,225 @@
+    "mg",  // Malagasy
+    "mh",  // Marshallese
+    "mi",  // Maori
+    "mk",  // Macedonian
+    "ml",  // Malayalam
+    "mn",  // Mongolian
+    "mr",  // Marathi
+    "ms",  // Malay
+    "mt",  // Maltese
+    "my",  // Burmese
+    "na",  // Nauru
+    "nb",  // "Bokmål, Norwegian; Norwegian Bokmål"
+    "nd",  // "Ndebele, North; North Ndebele"
+    "ne",  // Nepali
+    "ng",  // Ndonga
+    "nl",  // Dutch; Flemish
+    "nn",  // "Norwegian Nynorsk; Nynorsk, Norwegian"
+    "no",  // Norwegian
+    "nr",  // "Ndebele, South; South Ndebele"
+    "nv",  // Navajo; Navaho
+    "ny",  // Chichewa; Chewa; Nyanja
+    "oc",  // Occitan (post 1500)
+    "oj",  // Ojibwa
+    "om",  // Oromo
+    "or",  // Oriya
+    "os",  // Ossetian; Ossetic
+    "pa",  // Panjabi; Punjabi
+    "pi",  // Pali
+    "pl",  // Polish
+    "ps",  // Pushto; Pashto
+    "pt",  // Portuguese
+    "qu",  // Quechua
+    "rm",  // Romansh
+    "rn",  // Rundi
+    "ro",  // Romanian; Moldavian; Moldovan
+    "ru",  // Russian
+    "rw",  // Kinyarwanda
+    "sa",  // Sanskrit
+    "sc",  // Sardinian
+    "sd",  // Sindhi
+    "se",  // Northern Sami
+    "sg",  // Sango
+    "si",  // Sinhala; Sinhalese
+    "sk",  // Slovak
+    "sl",  // Slovenian
+    "sm",  // Samoan
+    "sn",  // Shona
+    "so",  // Somali
+    "sq",  // Albanian
+    "sr",  // Serbian
+    "ss",  // Swati
+    "st",  // "Sotho, Southern"
+    "su",  // Sundanese
+    "sv",  // Swedish
+    "sw",  // Swahili
+    "ta",  // Tamil
+    "te",  // Telugu
+    "tg",  // Tajik
+    "th",  // Thai
+    "ti",  // Tigrinya
+    "tk",  // Turkmen
+    "tl",  // Tagalog
+    "tn",  // Tswana
+    "to",  // Tonga (Tonga Islands)
+    "tr",  // Turkish
+    "ts",  // Tsonga
+    "tt",  // Tatar
+    "tw",  // Twi
+    "ty",  // Tahitian
+    "ug",  // Uighur; Uyghur
+    "uk",  // Ukrainian
+    "ur",  // Urdu
+    "uz",  // Uzbek
+    "ve",  // Venda
+    "vi",  // Vietnamese
+    "vo",  // Volapük
+    "wa",  // Walloon
+    "wo",  // Wolof
+    "xh",  // Xhosa
+    "yi",  // Yiddish
+    "yo",  // Yoruba
+    "za",  // Zhuang; Chuang
+    "zh",  // Chinese
+    "zu",  // Zulu
+  ]
+
+  static let languageSynonyms: [String: String] = [
+    "in": "id",
+    "iw": "he",
+    "ji": "yi",
+    "jw": "jv",
+    "mo": "ro",
+  ]
+
+  static func answerForLangaugeCode(_ languageCode: String?) -> Int {
+    let suspendedMetricValue = Int.max - 1
+    guard let languageCode = languageCode,
+      case let primaryLanguage = languageSynonyms[languageCode, default: languageCode],
+      let answer = acceptedLanguages.firstIndex(of: primaryLanguage)
+    else {
+      // Suspend metric if primary language not detected or language isn't accepted
+      return suspendedMetricValue
+    }
+    return answer
+  }
+
+  public static func recordPrimaryLanguageP3A() {
+    let primaryLanguageHistogramName = "Brave.Core.PrimaryLang"
+    let answer = answerForLangaugeCode(Locale.current.languageCode)
```

Comment:
Does it work correctly for longer language codes like `pt_BR` `pt_PT`? 
Is it a gap we should discuss?

Reply:
The language code for those would both be `pt`, the extras (BR/PT) are the region code